client/state/sharing/services/selectors.js

```diff
@@ -9,6 +21,58 @@ export function getKeyringServices( state ) {
 }
 
 /**
+ * Returns an object of service objects with the specified type.
+ *
+ * @param  {Object} state Global state tree
+ * @param  {String} type  Type of service. 'publicize' or 'other'.
+ * @return {Array}        Keyring services, if known.
+ */
+export function getKeyringServicesByType( state, type ) {
+	return filter( getKeyringServices( state ), { type } );
+}
+
+/**
+ * Returns an object of eligible service objects with the specified type.
+ *
+ * @param  {Object} state  Global state tree
+ * @param  {String} type   Type of service. 'publicize' or 'other'.
+ * @return {Array}         Keyring services, if known.
+ */
+export function getEligibleKeyringServices( state, type ) {
+	const siteId = getSelectedSiteId( state ),
```

Comment:
This is much improved, but I still think there's value in being able to call this selector for any site, not necessarily the currently selected site. That is to say, I think `siteId` should remain as an argument to the function, and that we call `getSelectedSiteId` from the `<ServicesGroup />` `connect` function instead.